src/map.js

```diff
@@ -649,7 +647,30 @@ var map = function (arg) {
         height: m_height
       };
     }
-    m_this.resize(0, 0, arg.width, arg.height);
+    // store the original center and restore it after the resize
+    var oldCenter = m_this.center();
```

Comment:
perhaps my intentions were not clear. I was wondering if there is a strong reason to wrap these few lines of code as internal method. If not, then I am good with these changes :+1: